backend/src/main/java/com/google/blackswan/mock/SimpleRelatedDataGenerator.java

```diff
@@ -0,0 +1,119 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.blackswan.mock;
+
+import com.google.models.*;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.ArrayListMultimap;
+
+/** Singleton class to generate related data for a given anomaly. */
+public class SimpleRelatedDataGenerator implements RelatedDataGenerator {
+  private static final String TARGET_DIMENSION = "Ramen";
+  private static final String RELATED_METRIC = "Interest Over Time";
+  private static final String CONFIG_USERNAME = "catyu@";
+  private static final ImmutableMap<String, String> FILE_LOCATION = ImmutableMap.of(
+                                                                      "Udon", "/udon-data.csv",
+                                                                      "Pho", "/pho-data.csv"
+                                                                    );
+
+  private static final SimpleRelatedDataGenerator INSTANCE = new SimpleRelatedDataGenerator();
+
+  private Multimap<String, DataInfo> relatedDataMap;
+  private Map<String, Map<Timestamp, Integer>> csvDataCache;
+
+  private SimpleRelatedDataGenerator() {
+    relatedDataMap = ArrayListMultimap.create();
+    csvDataCache = new HashMap<>();
+    prefillRelatedData(); // TODO: Remove when connected to datastore.
+  }
+
+  /** 
+   * Simulates action of querying configs from datastore and adding to 
+   * relatedDataMap. Currently, manually adds these relationship.
+   */
+  private void prefillRelatedData() {
+    // TODO: Replace with call for querying configs from datastore.
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Udon", CONFIG_USERNAME));
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Pho", CONFIG_USERNAME));
+    // Multimap looks like this right now: {{Ramen,{Udon, Pho}}, ...}.
+  }
+
+  public List<RelatedData> getRelatedData(String metricName, String dimensionName, 
+      Timestamp startTime, Timestamp endTime) {
+    List<RelatedData> relatedDataList = new ArrayList<RelatedData>();
+    // TODO: Currently only tries to find related data for dimension, ex. Ramen -> Pho, 
+    //       does not include logic for metric, should be similar.
+    if (relatedDataMap.containsKey(dimensionName)) {
+      for (DataInfo relatedTopic : relatedDataMap.get(dimensionName)) {
+        Map<Timestamp, Integer> topicDataPoints = 
+            getTopicDataPoints(relatedTopic.getDimensionName());
+
+        Map<Timestamp, MetricValue> dataPointsPlot = 
+            getDataPointsInRange(topicDataPoints, startTime, endTime);
```

Comment:
optional: maybe you could call `getTopicDataPoints` directly in `getDataPointsInRange`, since you're not using the constant `topicDataPoints` anywhere else?
as in, instead of passing it in, you could just call `getTopicDataPoints` in the function instead of in this function
maybe I understood the code wrong tho and you need to pass it in

Reply:
i think that's a great suggestion. i have refactored it, but i guess one tiny downside is that `getDataPointsInRange` cannot be a `static` method any more since it needs to call `getTopicDataPoints`. but i think that's a small trade off :)